### PR TITLE
DROOLS-4601 : Escape characters added to code after editing comma separated list in guided rule editor

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-4601

### CODE REVIEW

**Comment by Rikkola** [comment]
@jomarko @manstis if you two could be kind enough to review.

**Comment by jomarko** [comment]
No comments to the code, however it seems to me quotes are added in some strange way now:

- Screenshot from 2019-10-09 10-06-30
- 
Screenshot from 2019-10-09 10-06-39

**Comment by Rikkola** [comment]
@jomarko that is actually how master also works without this PR. I dare to say quotes haven't worked in that setup before. I'll try to see if there if there is a way to add support for this without breaking backwards compatibility.

**Comment by jomarko** [comment]
@Rikkola ok, sorry I need to clarify. Are you saying that you are fixing just case, when user put in values without double quotes and then after reopening still no double quotes added in UI? just in the source code? And are you saying that if user put in values already with double quotes it is not working correctly even without your PR?

**Comment by Rikkola** [comment]
Yes so even without this change, if you used double quotes like this `a, "b,c", d` it does a broken split between b and c. It looks like quotes do not work with list in GRE and never have.

This PR fixes the problem when you have a list `a,b,c` and it is changed to `"a", "b", "c"` in source. However if you reopen the rule textbox content is `\"a\", \"b\", \"c\"` if you save for a second time a new set of quotes is added and you can keep adding quotes this way infinitely. This fix keeps the content as it is, even if it is broken.

**Comment by Rikkola** [comment]
I'm not sure if that helped. The explanation is not the clearest :)

**Comment by Rikkola** [comment]
@jomarko what are your feelings on us making a separate ticket for the issue you found? That is how it worked before and it looks like the " quotes are not used to group the content inside of them to produce a single "a, b" element. There is a small possiblity that we break backward compatibility for any items that use " character in as an option. Although I it is very unlikely that anyone uses it.

**Comment by jomarko** [comment]
New issue reported as https://issues.jboss.org/browse/RHPAM-2457

**Review by jomarko — APPROVED** [review]

**Comment by jomarko** [comment]
@manstis could you please review and possibly merge?

File: drools-workbench-models/drools-workbench-models-commons/src/test/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceUnmarshallingTest.java [hunk 1]
```diff
@@ -9563,4 +9563,27 @@ public void testNotMatchesAndNotSoundsLikeDeclaredInDsl() {
...
                                      RuleModelDRLPersistenceImpl.getInstance().marshal(model));
     }
+
+    @Test
+    public void unmarshalStringListsCorrectly() {
+        final String drl = "package org.mortgages;\n" +
+                "rule \"aaa\"\n" +
+                "\tdialect \"mvel\"\n" +
+                "\twhen\n" +
+                "\t\tApplicant( applicantName : name in ( \"a\", \"b\", \"c\" ) )\n" +
```
Comment (manstis) [inline drools-workbench-models/drools-workbench-models-commons/src/test/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceUnmarshallingTest.java:9573]: @Rikkola Any chance we could have a test for strings containing commas; e.g. `"a", "b", "c,d", "e"`?

Comment (jomarko) [reply drools-workbench-models/drools-workbench-models-commons/src/test/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceUnmarshallingTest.java:9573]: @manstis it doesn't work, see my https://github.com/kiegroup/drools/pull/2565#issuecomment-542639075

**Review by manstis — APPROVED** [review]
Approving on the basis of https://github.com/kiegroup/drools/pull/2565#discussion_r335906561 (a new JIRA has been created for the test I requested be added to this PR).

**Pull request merged**